src/logger/logger.d.ts

```diff
@@ -0,0 +1,35 @@
+// Core log management functionality. For writing log output from code, see
+// logging.d.ts.
+//
+
+declare module Logger {
+  class Log {
+    constructor();
+    debug(tag:string, msg:string) : void;
+    info(tag:string, msg:string) : void;
+    warn(tag:string, msg:string) : void;
+    error(tag:string, msg:string) : void;
+  }
+
+  // The data structure for a logged message.
+  interface Message {
+    timestamp :Date; // the timestamp the log was called (in core runtime).
+    level :string; // one of D=Debug, I=Info, W=Warning, E=Error
+    tag :string; // any string, used for viewing specific module logs.
+    message :string; // the actual log message.
+  }
+
+  // This is the real internal interface available to other core modules/the
+  // top level environment.
+  function makeMessage(level:string, tag:string, msg:string)
```

Comment:
can this happen in a second pull request? It should ideally be done with getLogs, and it's a whole other API that we'll want to come to consensus on.